caluma/extensions/events/form.py

```diff
@@ -0,0 +1,122 @@
+import csv
+import io
+import logging
+
+from django.db import transaction
+from django.db.models.signals import post_save
+
+from caluma.caluma_core.events import on
+from caluma.caluma_form import models as caluma_form_models
+
+from ..settings import settings
+
+logger = logging.getLogger(__name__)
+
+
+@on(post_save, sender=caluma_form_models.AnswerDocument, raise_exception=True)
+@transaction.atomic
+def update_table_summary(instance, *args, **kwargs):
+    """
+    Update table summary.
+
+    If a question within a table's row form has the `meta` attributes
+    `summary-question` and `summary-mode` set, then upon save, a summary
+    will be written, as follows:
+
+    In the parent document (the one containing the table answer), the
+    referenced summary question is updated to contain a summary, calculated
+    according to the given summary mode.
+
+    Currently, only `csv` is allowed as a summary mode, but the code can be extended
+    to support sums, averages, etc. as needed.
+    """
+    question = instance.answer.question
+    main_document = instance.answer.document
+
+    summary_question = question.meta.get("summary-question")
+    summary_mode = question.meta.get("summary-mode")
+
+    if not summary_question and not summary_mode:
+        # no summary requested
+        return
+
+    msg = f"Updating table summary: tq={instance.answer.question_id}"
+    logger.debug(msg)
```

Comment:
Why do we log like this instead of 

```suggestion
    logger.debug(f"Updating table summary: tq={instance.answer.question_id}")
```

Reply:
We don't use f'strings in log statements because of [G004](https://docs.astral.sh/ruff/rules/logging-f-string/). I've updated it to use the default machanism from the `logging` module.

Reply:
Right - I hate the syntax, but oh well. Doing the fstring separately to avoid G004 did kind of avoid the purpose tho :-)